identity-wallet/src/state/profile_settings/reducers/update_sorting_preference.rs

```diff
@@ -0,0 +1,462 @@
+        let (mut credentials_with_date_issued, mut credentials_without_date_issued) = credentials
+            .into_iter()
+            // splits the `credentials` into two vectors.
+            .partition::<Vec<_>, _>(|credential| !credential.metadata.date_issued.is_empty());
+
+        credentials_without_date_issued.sort_by(name_az);
+        credentials_with_date_issued.append(&mut credentials_without_date_issued);
+        // assign the value reconstructed credentials back to `credentials`.
+        credentials = credentials_with_date_issued
+    };
+
+    // current_user_prompt is not set to None,
+    // as this reducer is often used in combination with reducers that need to send a user_prompt to the frontend.
+    Ok(AppState { credentials, ..state })
+}
+
+pub async fn sort_connections(state: AppState, _action: Action) -> Result<AppState, AppError> {
+    let mut connections: Vec<Connection> = state.connections.0.clone();
+    let preferences: Preferences<ConnectionSortMethod> = state.profile_settings.sorting_preferences.connections.clone();
+
+    let names: Vec<String> = connections.iter().map(|connection| connection.name.clone()).collect();
+    let sorted_names = sort(names, state.profile_settings.locale.clone());
+
+    let name_az = |a: &Connection, b: &Connection| {
+        let cmp_a = sorted_names.iter().position(|name| name == &a.name);
+        let cmp_b = sorted_names.iter().position(|name| name == &b.name);
+        cmp_a.cmp(&cmp_b)
+    };
+
+    match preferences.sort_method {
+        ConnectionSortMethod::NameAZ => connections.sort_by(name_az),
+        ConnectionSortMethod::FirstInteractedNewOld => {
+            connections.sort_by(|a: &Connection, b: &Connection| a.first_interacted.cmp(&b.first_interacted))
+        }
+        ConnectionSortMethod::LastInteractedNewOld => {
+            connections.sort_by(|a: &Connection, b: &Connection| a.last_interacted.cmp(&b.last_interacted))
+        }
+    };
+
+    if preferences.reverse {
+        connections.reverse();
+    }
+
+    // current_user_prompt is not set to None,
+    // as this reducer is often used in combination with reducers that need to send a user_prompt to the frontend.
+    Ok(AppState {
+        connections: Connections(connections),
+        ..state
+    })
+}
+
+// Helpers //
+
+/// Sort a Vector of words alphabetically, taking into account the locale of the words:
+/// standard `.sorted()` -> ["Zambia", "enlever", "zoo", "énigme"]
+/// `sort(list, locale)` -> ["énigme", "enlever", "Zambia", "zoo"]
+/// Also making it Unicode compatible, using NFC-normalized form.
+pub fn sort(list: Vec<String>, locale: Locale) -> Vec<String> {
+    // https://github.com/unicode-org/icu4x/tree/main/components/collator#examples
+
+    // Convert locale from AppState to icu_locid::Locale
+    let locale: &str = locale.into();
+    let locale: icu::locid::Locale = locale.replace('_', "-").parse().unwrap();
```

Comment:
There's no need to replace here I believe. Both formats will be parsed equally:
```rust

#[test]
fn test() {
    let locale1: icu::locid::Locale = "en-GB".parse().unwrap();
    let locale2: icu::locid::Locale = "en_GB".parse().unwrap();

    assert_eq!(locale1, locale2);
}
```

Reply:
updated